### ISSUE TITLE
move single renamed EXDEVed files

### ISSUE THREAD

**Issue by trapexit** [trapexit/mergerfs#217]
If a rename will result in a EXDEV with a path preserving policy: attempt to replicate what `mv` would do but only for single files. Should cover a decent percentage of situations. If not perhaps look into full `mv` behavior replication.

**Comment by stormwatch** [comment]
I was going to request for mv to actually use the same device whenever possible. Is this somehow related?

**Comment by trapexit** [comment]
No. This is to completely hide the underlying fact that a rename across drives can result in EXDEV when using a path preserving policy. IE. It'd do what `mv` is doing when the situation arises so that apps that aren't written well (many of them) hopefully won't freak out as they may now when they receive an EXDEV error.

Define 'whenever possible.' mergerfs[ already tries really hard to rename/link](https://github.com/trapexit/mergerfs#rename--link) on the same device given the policy restrictions. It's actually the most complicated code within mergerfs. If a non-path preserving policy is used then it should never not succeed. And with path preserving you can use `ignorepponrename` to ignore the path preserving only on rename and link.
